realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/PartialSyncTests.java

```diff
@@ -53,26 +49,24 @@ public void invalidQuery() {
         final Realm realm = Realm.getInstance(partialSyncConfig);
         looperThread.closeAfterTest(realm);
 
-        RealmResults<AllTypes> query = realm.where(AllTypes.class).findAllAsync();
-        query.addChangeListener(new OrderedRealmCollectionChangeListener<RealmResults<AllTypes>>() {
-            @Override
-            public void onChange(RealmResults<AllTypes> allTypes, OrderedCollectionChangeSet changeSet) {
-                switch (callbacks.incrementAndGet()) {
-                    case 1:
-                        assertEquals(OrderedCollectionChangeSet.State.INITIAL_INCOMPLETE, changeSet.getState());
-                        break;
-
-                    case 2:
-                        assertEquals(OrderedCollectionChangeSet.State.ERROR, OrderedCollectionChangeSet.State.ERROR);
-                        assertTrue(changeSet.getError() instanceof IllegalArgumentException);
-                        Throwable iae = changeSet.getError();
-                        assertTrue(iae.getMessage().contains("ERROR: realm::QueryParser: Comparison operator missing"));
-                        looperThread.testComplete();
-                        break;
-
-                    default:
-                        fail("Unexpected state: " + changeSet.getState());
-                }
+        // Backlinks not yet supported: https://github.com/realm/realm-core/pull/2947
+        RealmResults<AllJavaTypes> query = realm.where(AllJavaTypes.class).equalTo("objectParents.fieldString", "Foo").findAllAsync();
+        query.addChangeListener((results, changeSet) -> {
+            switch (callbacks.incrementAndGet()) {
+                case 1:
+                    assertEquals(OrderedCollectionChangeSet.State.INITIAL, changeSet.getState());
```

Comment:
Do we have the guarantee that we `INITIAL` state will be triggered and not directly `ERROR`?

Reply:
Yes, it is based on the Object Store implementation which should be deterministic about this